site-src/geps/gep-1619.md

```diff
@@ -0,0 +1,89 @@
+| Google Kubernetes Engine | ? | ? | ? | ? |
+| HAProxy Ingress | HAProxy | Cookie-Based | Stick Tables | HAProxy implements cookie-based stateful sessions via the [forwardfor option](https://docs.haproxy.org/2.6/configuration.html#4-option%20forwardfor). [Stick tables](https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#4-stick-table) implement session affinity via client attributes. |
+| HashiCorp Consul | Envoy | [Cookie-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/cookie/v3/cookie.proto), [Header-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/header/v3/header.proto) | Consistent Hashing via [Maglev](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#maglev) and [Ring Hash](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#ring-hash)| N/A |
+| Istio | Envoy | [Cookie-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/cookie/v3/cookie.proto), [Header-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/header/v3/header.proto) | Consistent Hashing via [Maglev](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#maglev) and [Ring Hash](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#ring-hash)| N/A |
+| Kong | Kong | Cookie-Based | ? | Kong has the [session plugin](https://docs.konghq.com/hub/kong-inc/session/) which manages persistent sessions with cookie-based stateful sessions. |
+| Kuma | Envoy | [Cookie-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/cookie/v3/cookie.proto), [Header-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/header/v3/header.proto) | Consistent Hashing via [Maglev](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#maglev) and [Ring Hash](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/load_balancers#ring-hash)| N/A |
+| LiteSpeed Ingress Controller | ? | ? | ? | ? |
+| NGINX Kubernetes Gateway | Nginx | ? | ? | ? |
+| Traefik |  Traefik Proxy | [Cookie-Based](https://doc.traefik.io/traefik/routing/services/#sticky-sessions) | No | [Docs](https://traefik.io/glossary/what-are-sticky-sessions/) mention consistent hashing, but Traefik [only offers round robin](https://doc.traefik.io/traefik/routing/services/#load-balancing) (not consistent). |
+
+### Open Questions
+- What are the needs and use cases from end users for specifying session persistence?
```

Comment:
Do we need to consider how to drain established sessions?  For example, an application developer wants to upgrade backends without disrupting any established sessions but rather wait for the sessions to expire.


example of this use case -  https://github.com/jcmoraisjr/haproxy-ingress/issues/137

Reply:
Yes, Istio implementation also includes draining ( using k8s new status ) - for the rollout of a new version it is better to gradually set traffic to 0% on old version and let the connection count drop over how long you need. 

However for autoscale down - drain support and a lot of tricks are needed ( stop hook, etc). Would be great to have a common doc, I assume all vendors will have the same problems - but may be a separate discussion.

Reply:
In [GEP 1709](https://gateway-api.sigs.k8s.io/geps/gep-1709/) we added an explicit [TODO Section](https://gateway-api.sigs.k8s.io/geps/gep-1709/#todo) as a list of things to follow up on in later PRs, and also as a list of things we felt would need to block moving the GEP to `implementable`. I would recommend using that same approach here to reduce how much more we need to cover in this single PR.

Reply:
Adding to a TODO section.